engine/runtime/src/main/java/org/enso/interpreter/runtime/ResourceManager.java

```diff
@@ -124,7 +129,7 @@ public ManagedResource register(Object object, Object function) {
     }
     if (workerThread == null || !workerThread.isAlive()) {
       worker.setKilled(false);
-      workerThread = context.getEnvironment().createThread(worker);
+      workerThread = context.getEnvironment().createSystemThread(worker);
```

Comment:
We need thread to read _"finalizable"_ references from `ReferenceQueue` - however it can be "system thread" - a thread that is not allowed to execute any guest language code.